test/SlowTests/Client/Subscriptions/ConcurrentSubscriptionsTests.cs

```diff
@@ -171,45 +171,59 @@ await using (var Subscription2 = store.Subscriptions.GetSubscriptionWorker(new S
                     var con2Docs = new List<string>();
 
                     var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    var mre = new ManualResetEvent(false);
-
-                    var _ = Subscription2.Run(x =>
+                    try
                     {
-                        foreach (var item in x.Items)
+                        var mre = new ManualResetEvent(false);
+
+                        var _ = Subscription2.Run(x =>
                         {
-                            con2Docs.Add(item.Id);
-                        }
+                            foreach (var item in x.Items)
+                            {
+                                con2Docs.Add(item.Id);
+                            }
 
-                        mre.Set();
-                        tcs.Task.Wait();
-                    });
+                            mre.Set();
+                            tcs.Task.Wait();
```

Comment:
can we make it `async` and `await` this?

Reply:
done